Source/Data/Remotes/ParleyRemote.swift

```diff
@@ -4,9 +4,60 @@ import ObjectMapper
 import UIKit
 import Foundation
 
+
+public typealias ParleyRemoteInterceptor = ParleyRemoteRequestInterceptorProtocol & ParleyRemoteResponseInterceptorProtocol
+
+public protocol ParleyRemoteRequestInterceptorProtocol: AnyObject {
+    func httpRequestWillStart(with request: inout URLRequest) throws
+}
+
+public protocol ParleyRemoteResponseInterceptorProtocol: AnyObject {
+    func httpResponseReceived(response: HTTPURLResponse) throws
+}
+
+internal class ParleyRemoteInterceptorProxy: RequestInterceptor {
+
+    private let interceptors: [ParleyRemoteInterceptor]
+    
+    init(interceptors: [ParleyRemoteInterceptor]) {
+        self.interceptors = interceptors
+    }
+    
+    func adapt(_ urlRequest: URLRequest, for session: Session, completion: @escaping (Result<URLRequest, Error>) -> Void) {
+        guard interceptors.count > 0 else {
```

Comment:
Please change the guard to `!interceptors.isEmpty` checks, instead of `> 0` checks

However, if throwing isn't needed, a for loop will suffice and this check isn't needed